manager-rs/directory-service/src/database/backups.rs

```diff
@@ -0,0 +1,161 @@
+use sqlx::sqlite::{SqlitePool, SqliteRow};
+use sqlx::Error as SqlxError;
+use sqlx::{Sqlite, Transaction};
+
+use crate::data::error::DirectoryError;
+
+/// Creates the backup networks table if it does not exist already.
+/// Contains all networks that are used as a backup for this network
+pub async fn init_table(pool: &SqlitePool) -> Result<(), DirectoryError> {
+    sqlx::query(
+        "CREATE TABLE IF NOT EXISTS backups_directory_table (
+            user_id TEXT NOT NULL,
+            backup_network_id TEXT NOT NULL,
+            PRIMARY KEY (user_id, backup_network_id)
```

Comment:
You probably also want a [foreign key constraint](https://www.sqlite.org/foreignkeys.html) here to make sure the user and network ids are valid in their respective tables.

Reply:
Really good idea! I wasn't entirely sure how to handle this problem originally. One slight downside is that the directory service would then require all backup networks to be registered before a user could be added. Otherwise, we would need to decide between total or partial failure to add backup networks for a given user.

Reply:
In terms of an actual operational system, I think it makes sense to require that the backup network be registered in the directory before a user can be registered that depends on that backup network. I see how this could be more of an issue in testing though when we're bulk adding users, but I think it could be handled by just having the directory service itself add the test users instead of having each network add them. In general if you don't enforce this constraint, you application code now needs to handle the failure mode that the backup directory doesn't actually exist for a given user.